api/api_full.go

```diff
@@ -299,6 +299,39 @@ type MinerPower struct {
 	TotalPower power.Claim
 }
 
+type MinerInfo struct {
+	Owner                      address.Address // Must be an ID-address.
+	Worker                     address.Address // Must be an ID-address.
+	NewWorker                  address.Address // Must be an ID-address.
+	WorkerChangeEpoch          abi.ChainEpoch
+	PeerId                     peer.ID
```

Comment:
Could we modify the accesses to the peer ID instead of duplicating the entire structure?

Reply:
The main problem here is that the json marshaling is wrong for the struct from the specs actors. We could add json marshaling methods to the specs actors code, but that would then require us to reimport libp2p, which is the whole reason we're here (they wanted to not have to import libp2p for this one type)

Reply:
I meant modifying the accesses in Lotus (consumer of the specs struct), they don't seem to be that many and we seem to already do it in cases like:

Reply:
Sorry, I didn't state my motivation clearly before: this makes the code harder to understand (I don't even care about maintainability, it's not that complex of a structure and we basically just copy all of it). With this change, if I follow a miner field to better understand the actor I land here with zero documentation and even with the deceiving appearance that Lotus _owns_ this actor (and not the spec). Still, it is just one hop away from the actual source, so it's not a hard requirement in my book, I was trying to explore other alternatives, feel free to ignore.

Reply:
Yeah, I wasn't too happy about the duplication, but, eh, it gets the job done, and adding an "API type" didn't feel dreadful.